uportal-war/src/main/java/org/jasig/portal/rendering/StylesheetAttributeSource.java

```diff
@@ -133,4 +150,35 @@ public IStylesheetDescriptor getStylesheetDescriptor(HttpServletRequest request)
     }
     
     public abstract PreferencesScope getStylesheetPreferencesScope(HttpServletRequest request);
+
+    protected abstract Logger getLogger();
+
+    protected boolean shouldDoSpelEvaluationForAttributeValue(final String attributeValue) {
+        return this.portalSpELService != null && attributeValue != null && attributeValue.startsWith("${") && attributeValue.endsWith("}");
+    }
+
+    protected String getValueForSpelEvaluation(final String attributeValue) {
+        if (attributeValue != null && attributeValue.startsWith("${") && attributeValue.endsWith("}")) {
+            return attributeValue.substring(2, attributeValue.length()-1);
+        } else {
+            return attributeValue;
+        }
+    }
+
+    protected String doSpelEvaluationForAttributeValue(final WebRequest request, final String attributeValue) {
+        String result;
+        try {
+            final String valueForEvaluation = this.getValueForSpelEvaluation(attributeValue);
+            final Expression expression = this.portalSpELService.parseExpression(valueForEvaluation);
+            result = this.portalSpELService.getValue(expression, request, String.class);
```

Comment:
I used the separate parseExpression(...) and getValue(expression, ..) calls simply because I copied over code from elsewhere and that is how it was done there.  I wasn't really clear on the differences, if any, between getValue(Expression...) and getValue(String...).  Note that I also could have left out the try..catch in my code, but the desired behavior is to not let the exception bubble up, but rather to just return the expression as the value.